### PR TITLE
Try to convert to float only GNOMAD POPMAX values that are numerical

### LINKED ISSUE #4714: Failing to parse "GNOMADE_POPMAX"
**Describe the bug**

When running the latest version of Scout (still using the image built for the recently merged PR #4710), and loading a case, I get a flood of errors as such for a newly run hg002-sample:

```
2024-07-23 06:48:46 9a62864ff4ed scout.adapter.mongo.variant_loader[279] INFO Start inserting clinical snv variants into database
2024-07-23 06:48:46 9a62864ff4ed scout.parse.variant.transcript[279] ERROR Encountered error when parsing variant frequencies for transcript ENST00000379407: Failed to parse GNOMADE_POPMAX (could not convert string to float: 'nfe')
2024-07-23 06:48:46 9a62864ff4ed scout.parse.variant.transcript[279] WARNING Only decomposed/split and normalised VEP v90+ frequencies are supported
2024-07-23 06:48:46 9a62864ff4ed scout.parse.variant.transcript[279] ERROR Encountered error when parsing variant frequencies for transcript ENST00000379409: Failed to parse GNOMADE_POPMAX (could not convert string to float: 'nfe')
2024-07-23 06:48:46 9a62864ff4ed scout.parse.variant.transcript[279] WARNING Only decomposed/split and normalised VEP v90+ frequencies are supported
2024-07-23 06:48:46 9a62864ff4ed scout.parse.variant.transcript[279] ERROR Encountered error when parsing variant frequencies for transcript ENST00000379410: Failed to parse GNOMADE_POPMAX (could not convert string to float: 'nfe')
2024-07-23 06:48:46 9a62864ff4ed scout.parse.variant.transcript[279] WARNING Only decomposed/split and normalised VEP v90+ frequencies are supported
2024-07-23 06:48:46 9a62864ff4ed scout.parse.variant.transcript[279] ERROR Encountered error when parsing variant frequencies for transcript NM_001160184.2: Failed to parse GNOMADE_POPMAX (could not convert string to float: 'nfe')
2024-07-23 06:48:46 9a62864ff4ed scout.parse.variant.transcript[279] WARNING Only decomposed/split and normalised VEP v90+ frequencies are supported
2024-07-23 06:48:46 9a62864ff4ed scout.parse.variant.transcript[279] ERROR Encountered error when parsing variant frequencies for transcript NM_001367552.1: Failed to parse GNOMADE_POPMAX (could not convert string to float: 'nfe')
2024-07-23 06:48:46 9a62864ff4ed scout.parse.variant.transcript[279] WARNING Only decomposed/split and normalised VEP v90+ frequencies are supported
2024-07-23 06:48:46 9a62864ff4ed scout.parse.variant.transcript[279] ERROR Encountered error when parsing variant frequencies for transcript NM_001410697.1: Failed to parse GNOMADE_POPMAX (could not convert string to float: 'nfe')
2024-07-23 06:48:46 9a62864ff4ed scout.parse.variant.transcript[279] WARNING Only decomposed/split and normalised VEP v90+ frequencies are supported
2024-07-23 06:48:46 9a62864ff4ed scout.parse.variant.transcript[279] ERROR Encountered error when parsing variant frequencies for transcript NM_032129.3: Failed to parse GNOMADE_POPMAX (could not convert string to float: 'nfe')
2024-07-23 06:48:46 9a62864ff4ed scout.parse.variant.transcript[279] WARNING Only decomposed/split and normalised VEP v90+ frequencies are supported
2024-07-23 06:48:46 9a62864ff4ed scout.parse.variant.transcript[279] ERROR Encountered error when parsing variant frequencies for transcript XM_017002474.2: Failed to parse GNOMADE_POPMAX (could not convert string to float: 'nfe')
2024-07-23 06:48:46 9a62864ff4ed scout.parse.variant.transcript[279] WARNING Only decomposed/split and normalised VEP v90+ frequencies are supported
```

Loading it with an older version (our current production, 4.81), these errors did not appear for the same dataset.

```
2024-07-23 06:50:19 8f3b387d2df3 scout.adapter.mongo.variant_loader[717] INFO Start inserting clinical snv variants into database
2024-07-23 06:50:19 8f3b387d2df3 scout.adapter.mongo.variant_loader[717] WARNING Bulk insertion failed - attempting separate variant upsert for this bulk
2024-07-23 06:50:19 8f3b387d2df3 scout.adapter.mongo.variant_loader[717] WARNING Variant f2756eb7f1177c8655ce60020c18a9a2 already exists in database - modifying
2024-07-23 06:50:19 8f3b387d2df3 scout.adapter.mongo.variant_loader[717] WARNING Variant bf1753e63e63a90feba7ad212218b7f2 already exists in database - modifying
2024-07-23 06:50:19 8f3b387d2df3 scout.adapter.mongo.variant_loader[717] WARNING Variant 9944d6dd3007e380f81bf9d8d98606ad already exists in database - modifying
2024-07-23 06:50:19 8f3b387d2df3 scout.adapter.mongo.variant_loader[717] WARNING Variant dfb98f3c6d77cf45df4c681e4ca6e808 already exists in database - modifying
2024-07-23 06:50:19 8f3b387d2df3 scout.adapter.mongo.variant_loader[717] WARNING Bulk insertion failed - attempting separate variant upsert for this bulk
2024-07-23 06:50:19 8f3b387d2df3 scout.adapter.mongo.variant_loader[717] WARNING Variant f1f66f2f63442030f128a45c09f11439 already exists in database - modifying
2024-07-23 06:50:19 8f3b387d2df3 scout.adapter.mongo.variant_loader[717] WARNING Variant 53917a9e8be6c63a884bd9bf24e3c355 already exists in database - modifying
2024-07-23 06:50:19 8f3b387d2df3 scout.adapter.mongo.variant_loader[717] WARNING Bulk insertion failed - attempting separate variant upsert for this bulk
2024-07-23 06:50:19 8f3b387d2df3 scout.adapter.mongo.variant_loader[717] WARNING Variant 8e6480665a17c4d13e158709d1df1ecf already exists in database - modifying
2024-07-23 06:50:19 8f3b387d2df3 scout.adapter.mongo.variant_loader[717] WARNING Variant 4f34d5bac8aa2378395f7432ee06511f already exists in database - modifying
2024-07-23 06:50:19 8f3b387d2df3 scout.adapter.mongo.variant_loader[717] WARNING Bulk insertion failed - attempting separate variant upsert for this bulk
2024-07-23 06:50:19 8f3b387d2df3 scout.adapter.mongo.variant_loader[717] WARNING Variant cc029270d78f232ee1194c512d61121d already exists in database - modifying
2024-07-23 06:50:19 8f3b387d2df3 scout.adapter.mongo.variant_loader[717] WARNING Bulk insertion failed - attempting separate variant upsert for this bulk
2024-07-23 06:50:19 8f3b387d2df3 scout.adapter.mongo.variant_loader[717] WARNING Variant 51fc977278e474f7964fc5631fddf741 already exists in database - modifying
2024-07-23 06:50:19 8f3b387d2df3 scout.adapter.mongo.variant_loader[717] WARNING Bulk insertion failed - attempting separate variant upsert for this bulk
```

**To Reproduce**

I generated a fresh set of results file from our local pipeline using the hg002 dataset as input.

The error appears when loading the sample:

```
scout load case <base>/wgs/yaml/hg002-2.updated.yaml
```

**Additional context**

Looking at our VCF file by eye, the fields look like this (one column per transcript):

```
gnomADg                       rs1302711714                        rs1302711714             rs1302711714                        rs1302711714
gnomADg_AF_popmax             0.0308811                           0.0308811                0.0308811                           0.0308811
gnomADg_AF                    0.0203717                           0.0203717                0.0203717                           0.0203717
gnomADg_popmax                nfe                                 nfe                      nfe                                 nfe
```

Seems like the `gnomADg_popmax` colum is picked up as the max frequency, but actually the key is `gnomADg_AF_popmax`.

The error is produced both for exons and genome, i.e. keys `gnomADe_popmax` and `gnomADg_popmax`.

I suspect this is how the values look when output from VEP111, but I can inspect further exactly where they are coming from if that is relevant for this issue.

### PR DESCRIPTION
**Pull request by northwestwitch**
This PR adds a functionality or fixes a bug.
- Closes #4714 --> Try to parse to float only GNOMAD POPMAX values that are numerical

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. how to test it, possibly with real cases/data

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by Jakob37
- [x] tests executed by Jakob37

### CODE REVIEW

File: scout/parse/variant/transcript.py [hunk 1]
```diff
@@ -324,7 +324,7 @@ def set_variant_frequencies(transcript, entry):
                 continue
 
             value = entry[key]
-            if not value or value == ".":
+            if not value or value == "." or value.isalpha():
```
Comment (northwestwitch) [inline scout/parse/variant/transcript.py:327]: @Jakob37 something like this?

Comment (Jakob37) [reply scout/parse/variant/transcript.py:327]: I'll test it out!

**Review by Jakob37 — APPROVED** [review]
OK, this looks good! The errors are gone, and the case looks fine inside Scout. Thanks for fixing this 🙏

**Comment by northwestwitch** [comment]
> OK, this looks good! The errors are gone, and the case looks fine inside Scout. Thanks for fixing this 🙏

Great! It was just annoying messages but I think it was already parsing stuff right. Thanks for reporting and testing! 🚀

**Pull request merged**